narwhals/_arrow/dataframe.py

```diff
@@ -112,3 +117,37 @@ def select(
         pa = get_pyarrow()
         df = pa.Table.from_arrays([s._series for s in new_series], names=names)
         return self._from_dataframe(df)
+
+    def drop(self, *columns: str | Iterable[str]) -> Self:
+        return self._from_dataframe(self._dataframe.drop(list(flatten(columns))))
```

Comment:
pyarrow `Table.drop` has been renamed to `drop_columns` from version 12.0.0, however for now `drop`  is kept for backwards compatibility.